test/helpers/signers.js

```diff
@@ -0,0 +1,115 @@
+const { ethers } = require('hardhat');
+const { secp256k1 } = require('@noble/curves/secp256k1');
+const { secp256r1 } = require('@noble/curves/p256');
+const { generateKeyPairSync, privateEncrypt } = require('crypto');
+const { hashTypedData } = require('../../lib/@openzeppelin-contracts/test/helpers/eip712');
+
+const ensureLowerOrderS = (N, { s, recovery, ...rest }) => {
+  if (s > N / 2n) {
+    s = N - s;
+    recovery = 1 - recovery;
+  }
+  return { s, recovery, ...rest };
+};
+
+class BooleanSigner {
+  signTypedData() {
+    return '0x01';
+  }
+}
+
+class ERC7739Signer {
+  signTypedData(domain, types, contents) {
+    return this._signRaw(hashTypedData(domain, ethers.TypedDataEncoder.from(types).hash(contents)));
+  }
+}
+
+class ECDSASigner extends ERC7739Signer {
+  N = 0xfffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141n;
+
+  constructor() {
+    super();
+    this._privateKey = secp256k1.utils.randomPrivateKey();
+    this.publicKey = secp256k1.getPublicKey(this._privateKey, false);
+  }
+
+  _signRaw(messageHash) {
+    const sig = this._ensureLowerOrderS(secp256k1.sign(messageHash.replace(/0x/, ''), this._privateKey));
+    return ethers.Signature.from({
+      r: sig.r,
+      v: sig.recovery + 27,
+      s: sig.s,
+    }).serialized;
+  }
+
+  get EOA() {
+    return new ethers.Wallet(ethers.hexlify(this._privateKey));
+  }
+
+  _ensureLowerOrderS({ s, recovery, ...rest }) {
+    return ensureLowerOrderS(this.N, { s, recovery, ...rest });
+  }
+}
```

Comment:
As I said in the past, I think having this bastraction for ECDSA is a mistake. `ethers.Wallet` (or equivalent) should be all that we need. Having both coexist may just create confusion.

Additionally, there is a risk that we mess up somewhere in our implementation. It happened in the past !

I'd be favorable to implementing ethers.Signer compatible objects that supports signature using P256 and RSA ... But I would not touch ECDSA for which we basically have everything we need already